#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_fix_irqreturn_t.dpatch by Martijn van Brummelen <martijn@brumit.nl>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix for compiling the module (Closes: 545367)

@DPATCH@
diff -urNad acx100-20080210~/wlan_compat.h acx100-20080210/wlan_compat.h
--- acx100-20080210~/wlan_compat.h	2008-02-10 21:06:42.000000000 +0100
+++ acx100-20080210/wlan_compat.h	2009-11-05 10:07:50.000000000 +0100
@@ -218,7 +218,7 @@
 #define WLAN_PACKED	__attribute__ ((packed))
 
 /* Interrupt handler backwards compatibility stuff */
-#ifndef IRQ_NONE
+#if (! (defined(_LINUX_IRQRETURN_H) || defined(IRQ_NONE)))
 #define IRQ_NONE
 #define IRQ_HANDLED
 typedef void irqreturn_t;
